backend/core/llm_chain.py

```
@@ -3,490 +3,167 @@
+def cache_response(cache_key: str, response_chunks: List[str]):
+    if not cache_key or not ENABLE_CACHING: return
+    full_response = "".join(response_chunks)
+    _response_cache[cache_key] = {'response': full_response, 'timestamp': time.time()}
+
+def route_query_to_retrievers(query: str, retrievers: Dict[str, BaseRetriever]) -> List[BaseRetriever]:
+    query_lower = query.lower()
+    selected_names = set()
+    resume_keywords = ["experience", "job", "work", "skill", "resume", "cv", "company", "role", "hillman", "wisnet", "history"]
+    about_keywords = ["about", "background", "who is", "philosophy", "approach"]
+    illustration_keywords = ["art", "illustration", "drawing", "picture", "character", "design"]
+    if any(keyword in query_lower for keyword in resume_keywords): selected_names.add("resume")
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)